forwarder/src/lib.rs

```diff
@@ -2,22 +2,29 @@ use forwarder_utils::forwarder_prefix;
 use near_sdk::borsh::{self, BorshDeserialize, BorshSerialize};
 use near_sdk::json_types::U128;
 use near_sdk::{
-    assert_one_yocto, assert_self, env, ext_contract, near_bindgen, AccountId, Gas, PanicOnDefault,
-    Promise, PromiseOrValue, PublicKey,
+    assert_one_yocto, assert_self, env, ext_contract, near_bindgen, AccountId, Balance, Gas,
+    PanicOnDefault, Promise, PromiseOrValue, PublicKey,
 };
 use std::str::FromStr;
 
+const MINIMUM_BALANCE: Balance = 1_800_000_000_000_000_000_000_000;
 const MAX_FEE_PERCENT: u128 = 10;
 
-const FT_BALANCE_GAS: Gas = Gas(3_000_000_000_000);
 const CALCULATE_FEES_GAS: Gas = Gas(5_000_000_000_000);
-const FT_TRANSFER_GAS: Gas = Gas(10_000_000_000_000);
+const NEAR_DEPOSIT_GAS: Gas = Gas(5_000_000_000_000);
+const FT_TRANSFER_GAS: Gas = Gas(5_000_000_000_000);
 const FT_TRANSFER_CALL_GAS: Gas = Gas(30_000_000_000_000);
 const CALCULATE_FEES_CALLBACK_GAS: Gas = Gas(30_000_000_000_000);
 const FINISH_FORWARD_GAS: Gas = Gas(30_000_000_000_000);
 
 // Key is used for upgrading the smart contract.
 const UPDATER_PK: &str = "ed25519:BaiF3VUJf5pxB9ezVtzH4SejpdYc7EA3SqrKczsj1wno";
+// In case we get near as a token id it means we need to transfer native NEAR tokens.
+const NEAR: &str = "near";
+#[cfg(not(feature = "test"))]
+const WRAP_NEAR: &str = "wrap.near";
+#[cfg(feature = "test")]
+const WRAP_NEAR: &str = "wrap.test.near";
```

Comment:
is this wrapped near for testnet?

Reply:
No. It's for internal integration tests.

Reply:
Will we support wrapped near on the testnet?

Reply:
I'm not sure, but If we will, then we need to set the account id of wNEAR in the constructor of the forwarder contract.